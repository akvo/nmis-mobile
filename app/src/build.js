import { SERVER_URL } from '@env';
import build_json from './build.json';

const defaultBuildParams = {
  authenticationType: ['code_assignment', 'username', 'password'],
  ...build_json,
  serverURL: SERVER_URL,
};

export default defaultBuildParams;
