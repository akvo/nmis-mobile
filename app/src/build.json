{
  "authenticationType": ["code_assignment", "username", "password"],
  "serverURL": "http://192.168.21.252:8080",
  "debugMode": false,
  "dataSyncInterval": 300,
  "errorHandling": true,
  "loggingLevel": "verbose",
  "appVersion": "1.2.0",
  "lang": "en"
}
