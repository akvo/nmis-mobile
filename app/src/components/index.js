export { default as Stack } from './Stack';
export { default as Carousel } from './Carousel';
export { default as DividerLayout } from './DividerLayout';
