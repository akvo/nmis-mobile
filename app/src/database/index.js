export * from './conn';
export * from './query';
