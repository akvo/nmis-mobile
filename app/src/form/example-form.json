{
  "id": 519630048,
  "name": "Household",
  "version": "1.0.0",
  "question_group": [
    {
      "name": "Introduction",
      "question": [
        {
          "id": 364240038,
          "name": "Name of the data collector (Enumerator)",
          "order": 1,
          "type": "text",
          "required": true,
          "meta": true
        },
        {
          "id": 5196300481,
          "name": "Organisation",
          "order": 2,
          "type": "cascade",
          "required": true,
          "meta": false
        },
        {
          "id": 517690051,
          "name": "Household code",
          "order": 3,
          "type": "text",
          "required": true,
          "meta": true
        },
        {
          "id": 444670046,
          "name": "Are you willing to participate in the survey?",
          "order": 4,
          "type": "option",
          "required": true,
          "option": [
            {
              "id": 14485,
              "name": "Yes",
              "order": 1
            },
            {
              "id": 14486,
              "name": "No",
              "order": 2
            }
          ],
          "meta": false
        },
        {
          "id": 517600099,
          "name": "Number between 1 to 9 (decimal allowed)",
          "order": 5,
          "type": "number",
          "required": true,
          "meta": false,
          "rule": {
            "min": 1,
            "max": 9,
            "allowDecimal": true
          }
        }
      ]
    },
    {
      "name": "Demographics",
      "question": [
        {
          "id": 517600060,
          "name": "Take the GPS location of the Household",
          "order": 1,
          "type": "geo",
          "required": false,
          "center": {
            "lat": 9.145,
            "lng": 40.4897
          },
          "meta": true
        },
        {
          "id": 603100002,
          "name": "Is the household in an urban, peri-urban or rural setup?",
          "order": 4,
          "type": "option",
          "required": true,
          "option": [
            {
              "id": 14487,
              "name": "Rural",
              "order": 1
            },
            {
              "id": 14488,
              "name": "Peri-urban",
              "order": 2
            },
            {
              "id": 14489,
              "name": "Urban",
              "order": 3
            }
          ],
          "meta": false
        },
        {
          "id": 517600053,
          "name": "Age of the respondent",
          "order": 5,
          "type": "number",
          "required": true,
          "meta": false
        },
        {
          "id": 517600056,
          "name": "Respondent relationship to the household head",
          "order": 7,
          "type": "option",
          "required": true,
          "option": [
            {
              "id": 14492,
              "name": "Head",
              "order": 1
            },
            {
              "id": 14493,
              "name": "Spouse",
              "order": 2
            },
            {
              "id": 14494,
              "name": "Offspring (Above 18 years)",
              "order": 3
            },
            {
              "id": 14495,
              "name": "sibling",
              "order": 4
            },
            {
              "id": 14496,
              "name": "Parent",
              "order": 5
            }
          ],
          "meta": false
        },
        {
          "id": 608890003,
          "name": "What age are the members of the household?",
          "order": 11,
          "type": "multiple_option",
          "required": true,
          "option": [
            {
              "id": 14499,
              "name": "0-5",
              "order": 1
            },
            {
              "id": 14500,
              "name": "6-12",
              "order": 2
            },
            {
              "id": 14501,
              "name": "13-24",
              "order": 3
            },
            {
              "id": 14502,
              "name": "25-59",
              "order": 4
            },
            {
              "id": 14503,
              "name": "60+",
              "order": 5
            }
          ],
          "meta": false
        },
        {
          "id": 519660048,
          "name": "How many members are 0-5 years",
          "order": 12,
          "type": "number",
          "required": true,
          "dependency": [
            {
              "id": 608890003,
              "options": ["0-5"]
            }
          ],
          "meta": false
        },
        {
          "id": 517600058,
          "name": "How many members are between the age of 6-12 years?",
          "order": 13,
          "type": "number",
          "required": true,
          "dependency": [
            {
              "id": 608890003,
              "options": ["6-12"]
            }
          ],
          "meta": false
        },
        {
          "id": 519660050,
          "name": "how many member are between the age of 13-24 years?",
          "order": 14,
          "type": "number",
          "required": true,
          "dependency": [
            {
              "id": 608890003,
              "options": ["13-24"]
            }
          ],
          "meta": false
        },
        {
          "id": 600180100,
          "name": "How many members are between the age of 25-59 years?",
          "order": 15,
          "type": "number",
          "required": true,
          "dependency": [
            {
              "id": 608890003,
              "options": ["25-59"]
            }
          ],
          "meta": false
        },
        {
          "id": 519660054,
          "name": "How many members are above 60 years?",
          "order": 16,
          "type": "number",
          "required": true,
          "dependency": [
            {
              "id": 608890003,
              "options": ["60+"]
            }
          ],
          "meta": false
        },
        {
          "id": 519660052,
          "name": "Residential status of this household in the community",
          "order": 17,
          "type": "option",
          "required": true,
          "option": [
            {
              "id": 14504,
              "name": "Permanent",
              "order": 1
            },
            {
              "id": 14505,
              "name": "Seasonal migrant",
              "order": 2
            },
            {
              "id": 14506,
              "name": "Nomadic",
              "order": 3
            },
            {
              "id": 14507,
              "name": "Tenant",
              "order": 4
            },
            {
              "id": 14508,
              "name": "Refugee/IDPs",
              "order": 5
            }
          ],
          "meta": false,
          "extra": {
            "allowOther": true
          }
        }
      ]
    }
  ]
}
