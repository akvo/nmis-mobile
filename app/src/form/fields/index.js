export { default as TypeImage } from './TypeImage';
export { default as TypeInput } from './TypeInput';
export { default as TypeDate } from './TypeDate';
export { default as TypeOption } from './TypeOption';
export { default as TypeMultipleOption } from './TypeMultipleOption';
