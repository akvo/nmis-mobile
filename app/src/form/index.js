export { default as Webform } from './Webform';
