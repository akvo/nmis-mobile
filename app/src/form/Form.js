import React from 'react';
import { BaseLayout } from '../components';

const Form = ({ navigation, route }) => {
  const goBack = () => {
    navigation.navigate('FormAction', { ...route?.params });
  };
  return (
    <BaseLayout title={route?.params?.name} back={goBack}>
      <BaseLayout.Content data={[]} />
    </BaseLayout>
  );
};

export default Form;
