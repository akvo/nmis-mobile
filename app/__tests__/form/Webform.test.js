import React from 'react';
import renderer from 'react-test-renderer';
jest.useFakeTimers();
import { Webform } from '../../src/form';

describe('Webform component', () => {
  test('renders correctly', () => {
    const tree = renderer.create(<Webform />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
