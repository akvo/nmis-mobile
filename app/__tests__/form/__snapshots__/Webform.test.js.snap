// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Webform component should render component correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#f9fafb",
      "flex": 1,
      "flexDirection": "column",
      "flexWrap": "wrap",
    }
  }
  testID="stack-container"
>
  <RCTScrollView
    style={
      {
        "width": "100%",
      }
    }
  >
    <View>
      <View
        style={
          {
            "backgroundColor": "#f9fafb",
            "flex": 1,
            "flexDirection": "row",
            "flexWrap": "wrap",
          }
        }
        testID="stack-container"
      >
        <View
          style={
            {
              "flex": 1,
            }
          }
        >
          <View
            style={
              {
                "marginBottom": 20,
              }
            }
          >
            <View>
              <View
                style={
                  {
                    "borderBottomWidth": 0.5,
                    "borderColor": "grey",
                    "borderTopWidth": 0.5,
                    "paddingHorizontal": 20,
                    "paddingVertical": 15,
                  }
                }
              >
                <Text
                  accessibilityRole="text"
                  style={
                    {
                      "color": "#242424",
                      "fontSize": 14,
                      "fontWeight": "bold",
                    }
                  }
                >
                  Group 1
                </Text>
              </View>
              <Text
                accessibilityRole="text"
                style={
                  {
                    "color": "#242424",
                    "fontSize": 14,
                    "paddingHorizontal": 20,
                    "paddingVertical": 15,
                  }
                }
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi dapibus.
              </Text>
            </View>
            <View
              style={
                {
                  "paddingHorizontal": 10,
                  "paddingVertical": 10,
                }
              }
            >
              <Text
                accessibilityRole="text"
                style={
                  {
                    "color": "#242424",
                    "fontSize": 14,
                    "fontWeight": 600,
                    "marginBottom": 8,
                    "paddingHorizontal": 10,
                  }
                }
                testID="field-label"
              >
                Name
              </Text>
              <View
                style={
                  {
                    "paddingHorizontal": 10,
                    "width": "100%",
                  }
                }
                testID="RNE__Input__view-wrapper"
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "alignItems": "center",
                      "borderBottomWidth": 0.5,
                      "borderColor": "grey",
                      "borderRadius": 5,
                      "borderWidth": 0.5,
                      "flexDirection": "row",
                      "paddingHorizontal": 10,
                      "transform": [
                        {
                          "translateX": 0,
                        },
                      ],
                    }
                  }
                >
                  <TextInput
                    editable={true}
                    onChangeText={[Function]}
                    placeholderTextColor="#86939e"
                    style={
                      {
                        "color": "#242424",
                        "flex": 1,
                        "fontSize": 18,
                        "minHeight": 40,
                      }
                    }
                    testID="type-input"
                    underlineColorAndroid="transparent"
                    value="John Doe"
                  />
                </View>
                <Text
                  style={
                    {
                      "color": "#ff190c",
                      "fontSize": 12,
                      "margin": 5,
                    }
                  }
                />
              </View>
            </View>
            <View
              style={
                {
                  "paddingHorizontal": 10,
                  "paddingVertical": 10,
                }
              }
            >
              <Text
                accessibilityRole="text"
                style={
                  {
                    "color": "#242424",
                    "fontSize": 14,
                    "fontWeight": 600,
                    "marginBottom": 8,
                    "paddingHorizontal": 10,
                  }
                }
                testID="field-label"
              >
                Birth Date
              </Text>
              <View
                style={
                  {
                    "paddingHorizontal": 10,
                    "width": "100%",
                  }
                }
                testID="RNE__Input__view-wrapper"
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "alignItems": "center",
                      "borderBottomWidth": 0.5,
                      "borderColor": "grey",
                      "borderRadius": 5,
                      "borderWidth": 0.5,
                      "flexDirection": "row",
                      "paddingHorizontal": 10,
                      "transform": [
                        {
                          "translateX": 0,
                        },
                      ],
                    }
                  }
                >
                  <TextInput
                    editable={true}
                    onPressIn={[Function]}
                    placeholderTextColor="#86939e"
                    showSoftInputOnFocus={false}
                    style={
                      {
                        "color": "#242424",
                        "flex": 1,
                        "fontSize": 18,
                        "minHeight": 40,
                      }
                    }
                    testID="type-date"
                    underlineColorAndroid="transparent"
                    value="1/1/1991"
                  />
                </View>
                <Text
                  style={
                    {
                      "color": "#ff190c",
                      "fontSize": 12,
                      "margin": 5,
                    }
                  }
                />
              </View>
            </View>
            <View
              style={
                {
                  "paddingHorizontal": 10,
                  "paddingVertical": 10,
                }
              }
            >
              <Text
                accessibilityRole="text"
                style={
                  {
                    "color": "#242424",
                    "fontSize": 14,
                    "fontWeight": 600,
                    "marginBottom": 8,
                    "paddingHorizontal": 10,
                  }
                }
                testID="field-label"
              >
                Age
              </Text>
              <View
                style={
                  {
                    "paddingHorizontal": 10,
                    "width": "100%",
                  }
                }
                testID="RNE__Input__view-wrapper"
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "alignItems": "center",
                      "borderBottomWidth": 0.5,
                      "borderColor": "grey",
                      "borderRadius": 5,
                      "borderWidth": 0.5,
                      "flexDirection": "row",
                      "paddingHorizontal": 10,
                      "transform": [
                        {
                          "translateX": 0,
                        },
                      ],
                    }
                  }
                >
                  <TextInput
                    editable={true}
                    keyboardType="numeric"
                    onChangeText={[Function]}
                    placeholderTextColor="#86939e"
                    style={
                      {
                        "color": "#242424",
                        "flex": 1,
                        "fontSize": 18,
                        "minHeight": 40,
                      }
                    }
                    testID="RNE__Input__text-input"
                    underlineColorAndroid="transparent"
                    value="32"
                  />
                </View>
                <Text
                  style={
                    {
                      "color": "#ff190c",
                      "fontSize": 12,
                      "margin": 5,
                    }
                  }
                />
              </View>
            </View>
            <View
              style={
                {
                  "paddingHorizontal": 10,
                  "paddingVertical": 10,
                }
              }
            >
              <Text
                accessibilityRole="text"
                style={
                  {
                    "color": "#242424",
                    "fontSize": 14,
                    "fontWeight": 600,
                    "marginBottom": 8,
                    "paddingHorizontal": 10,
                  }
                }
                testID="field-label"
              >
                Gender
              </Text>
              <View
                accessibilityRole="checkbox"
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": true,
                    "disabled": false,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                checkedIcon="dot-circle-o"
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "backgroundColor": "transparent",
                    "borderColor": "#ffffff",
                    "margin": 5,
                    "marginLeft": 10,
                    "marginRight": 10,
                    "padding": 10,
                  }
                }
                testID="RNE__CheckBox__Wrapper"
                uncheckedIcon="circle-o"
              >
                <View
                  style={
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text />
                  <Text
                    accessibilityRole="text"
                    style={
                      {
                        "color": "#43484d",
                        "fontWeight": "normal",
                        "marginLeft": 10,
                        "marginRight": 10,
                      }
                    }
                    testID="RNE__CheckBox__Title"
                  >
                    Male
                  </Text>
                </View>
              </View>
              <View
                accessibilityRole="checkbox"
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": false,
                    "disabled": false,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                checkedIcon="dot-circle-o"
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "backgroundColor": "transparent",
                    "borderColor": "#ffffff",
                    "margin": 5,
                    "marginLeft": 10,
                    "marginRight": 10,
                    "padding": 10,
                  }
                }
                testID="RNE__CheckBox__Wrapper"
                uncheckedIcon="circle-o"
              >
                <View
                  style={
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text />
                  <Text
                    accessibilityRole="text"
                    style={
                      {
                        "color": "#43484d",
                        "fontWeight": "normal",
                        "marginLeft": 10,
                        "marginRight": 10,
                      }
                    }
                    testID="RNE__CheckBox__Title"
                  >
                    Female
                  </Text>
                </View>
              </View>
            </View>
            <View
              style={
                {
                  "paddingHorizontal": 10,
                  "paddingVertical": 10,
                }
              }
            >
              <Text
                accessibilityRole="text"
                style={
                  {
                    "color": "#242424",
                    "fontSize": 14,
                    "fontWeight": 600,
                    "marginBottom": 8,
                    "paddingHorizontal": 10,
                  }
                }
                testID="field-label"
              >
                Last Education
              </Text>
              <View
                onLayout={[Function]}
                style={
                  {
                    "borderColor": "grey",
                    "borderRadius": 5,
                    "borderWidth": 0.5,
                    "justifyContent": "center",
                    "marginHorizontal": 10,
                    "paddingHorizontal": 10,
                    "paddingVertical": 2,
                  }
                }
              >
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessible={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "height": 35,
                      "justifyContent": "space-between",
                    }
                  }
                  testID="type-option-dropdown"
                >
                  <Text
                    style={
                      [
                        {
                          "flex": 1,
                          "fontSize": 16,
                          "writingDirection": "ltr",
                        },
                        undefined,
                        {},
                      ]
                    }
                  >
                    Select item
                  </Text>
                  <Image
                    source={1}
                    style={
                      {
                        "height": 20,
                        "tintColor": "gray",
                        "width": 20,
                      }
                    }
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      {
        "width": "100%",
      }
    }
  >
    <View
      accessibilityRole="tablist"
      onLayout={[Function]}
      style={
        [
          false,
          {
            "flexDirection": "row",
            "position": "relative",
          },
          undefined,
        ]
      }
    >
      <View
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "borderRadius": 2,
            },
            [
              {
                "borderRadius": 0,
                "flex": 1,
              },
              false,
              undefined,
            ],
            false,
          ]
        }
        testID="RNE_BUTTON_WRAPPER"
      >
        <View
          accessibilityRole="tab"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": true,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": "Back",
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onLayout={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="form-nav-btn-back"
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderBottomWidth": 0,
                "borderColor": "#2089dc",
                "borderRadius": 0,
                "borderTopColor": "grey",
                "borderTopWidth": 0.5,
                "borderWidth": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 8,
                "paddingHorizontal": 10,
              }
            }
          >
            <View
              style={
                {
                  "marginHorizontal": 5,
                  "overflow": "hidden",
                  "paddingBottom": 10,
                  "paddingTop": 9,
                }
              }
              testID="RNE__ICON__CONTAINER"
            >
              <View
                accessibilityRole="button"
                android_ripple={
                  {
                    "borderless": false,
                    "color": "rgba(0, 0, 0, 0.3)",
                    "radius": -5,
                  }
                }
                disabled={false}
                testID="RNE__ICON__CONTAINER_ACTION"
              >
                <View
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    }
                  }
                  testID="RNE__ICON"
                >
                  <Text />
                </View>
              </View>
            </View>
            <Text
              style={
                {
                  "color": "grey",
                  "fontSize": 14,
                  "fontWeight": "normal",
                  "marginLeft": -10,
                  "marginRight": -10,
                  "paddingBottom": 10,
                  "paddingHorizontal": 16,
                  "paddingTop": 9,
                  "paddingVertical": 2,
                  "textAlign": "center",
                }
              }
            >
              Back
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "borderRadius": 2,
            },
            [
              {
                "borderRadius": 0,
                "flex": 1,
              },
              false,
              undefined,
            ],
            false,
          ]
        }
        testID="RNE_BUTTON_WRAPPER"
      >
        <View
          accessibilityRole="tab"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": true,
              "expanded": undefined,
              "selected": false,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": "1/2",
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onLayout={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="form-nav-group-count"
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderBottomWidth": 0,
                "borderColor": "#2089dc",
                "borderRadius": 0,
                "borderTopColor": "grey",
                "borderTopWidth": 0.5,
                "borderWidth": 0,
                "flexDirection": "column",
                "justifyContent": "center",
                "padding": 8,
                "paddingHorizontal": 10,
              }
            }
          >
            <Text
              style={
                {
                  "color": "hsl(208, 8%, 63%)",
                  "fontSize": 14,
                  "fontWeight": "normal",
                  "marginLeft": -10,
                  "marginRight": -10,
                  "paddingBottom": 10,
                  "paddingHorizontal": 16,
                  "paddingTop": 9,
                  "paddingVertical": 8,
                  "textAlign": "center",
                }
              }
            >
              1/2
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "borderRadius": 2,
            },
            [
              {
                "borderRadius": 0,
                "flex": 1,
              },
              false,
              undefined,
            ],
            false,
          ]
        }
        testID="RNE_BUTTON_WRAPPER"
      >
        <View
          accessibilityRole="tab"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": false,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": "Next",
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onLayout={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="form-nav-btn-next"
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderBottomWidth": 0,
                "borderColor": "#2089dc",
                "borderRadius": 0,
                "borderTopColor": "grey",
                "borderTopWidth": 0.5,
                "borderWidth": 0,
                "flexDirection": "row-reverse",
                "justifyContent": "center",
                "padding": 8,
                "paddingHorizontal": 10,
              }
            }
          >
            <View
              style={
                {
                  "marginHorizontal": 5,
                  "overflow": "hidden",
                  "paddingBottom": 10,
                  "paddingTop": 9,
                }
              }
              testID="RNE__ICON__CONTAINER"
            >
              <View
                accessibilityRole="button"
                android_ripple={
                  {
                    "borderless": false,
                    "color": "rgba(0, 0, 0, 0.3)",
                    "radius": -5,
                  }
                }
                disabled={false}
                testID="RNE__ICON__CONTAINER_ACTION"
              >
                <View
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    }
                  }
                  testID="RNE__ICON"
                >
                  <Text />
                </View>
              </View>
            </View>
            <Text
              style={
                {
                  "color": "grey",
                  "fontSize": 14,
                  "fontWeight": "normal",
                  "marginLeft": -10,
                  "marginRight": -10,
                  "paddingBottom": 10,
                  "paddingHorizontal": 16,
                  "paddingTop": 9,
                  "paddingVertical": 2,
                  "textAlign": "center",
                }
              }
            >
              Next
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
